core/src/main/java/org/apache/iceberg/TableOperations.java

```diff
@@ -52,6 +52,12 @@
    * Implementations must check that the base metadata is current to avoid overwriting updates.
    * Once the atomic commit operation succeeds, implementations must not perform any operations that
    * may fail because failure in this method cannot be distinguished from commit failure.
+   * <p>
```

Comment:
Requirement for new exception detailed in api here

Reply:
Looks good!